src/AltinnCore/Common/Services/Implementation/DataStudioSI.cs

```diff
@@ -303,9 +329,62 @@ public async Task<Guid> SaveFormAttachment(string org, string app, int instanceO
                 string instanceDataAsString = JsonConvert.SerializeObject(instance);
 
                 File.WriteAllText(instanceFilePath, instanceDataAsString);
-            }          
+                return data;
+            }
+        }
+
+        /// <inheritdoc />
+        public async Task<DataElement> UpdateFormAttachment(string org, string app, int instanceOwnerId, Guid instanceGuid, Guid dataGuid, HttpRequest request)
+        {
+            string developer = AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext);
+            Guid dataId = Guid.NewGuid();
+            long filesize;
+            string pathToSaveTo = $"{_settings.GetTestdataForPartyPath(org, app, developer)}{instanceOwnerId}/{instanceGuid}/data";
+            string fileToWriteTo = $"{pathToSaveTo}/{dataId}";
+
+            if (!File.Exists(fileToWriteTo))
+            {
+                _logger.LogError("Cannot find file to update.");
```

Comment:
is just the logging enough here? Should we throw an exception such that it doesn't execute the rest of the code?

Reply:
throwing and catching file not found exception instead.